src/main/java/gift/controller/ProductController.java

```diff
@@ -0,0 +1,50 @@
+package gift.controller;
+
+import gift.dto.ProductRequestDto;
+import gift.dto.ProductResponseDto;
+import gift.entity.Product;
+import gift.service.ProductService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/products")
+public class ProductController {
+
+    private final ProductService productService;
+
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @PostMapping
+    public ResponseEntity<ProductResponseDto> createProduct(@RequestBody ProductRequestDto requestDto) {
+        return new ResponseEntity<>(productService.saveProduct(requestDto), HttpStatus.CREATED);
+
+    }
+
+    @GetMapping
+    public ResponseEntity<List<ProductResponseDto>> findAllProducts() {
+        return new ResponseEntity<>(productService.findAllProducts(), HttpStatus.OK);
```

Comment:
```suggestion
        return ResponseEntity.ok(productService.findAllProducts());
```
이런식으로 간결하게 사용할 수 있습니다:)